src/bot/mod.rs

```diff
@@ -56,12 +57,22 @@ pub async fn run_bot(client: &Client, dispatcher: Arc<DownloadDispatcher>) -> Re
             continue;
         }
 
-        match handle_message(message, &client, dispatcher.clone()).await {
-            Ok(_) => {}
-            Err(e) => {
-                error!("Error occurred while handling message: {:?}", e);
+        let dispatcher = dispatcher.clone();
+        let client = client.clone();
+        tokio::spawn(async move {
```

Comment:
It's possible to accomplish without timing out the whole message handling process;
Nevertheless, it's more logical to timeout the video handling itself, considering possible future dialogue interface utilization (it would likely lead to full rewrite of the bot but anyway...)